Add support for .repo files (#1264058)

bcl opened the pull request:
This adds the --repo command which can be added multiple times to point
to dnf .repo files.

--enablerepo and --disablerepo can be used multiple times to control
which repos from the .repo files are actually used for the boot.iso
creation.

--repo can be used instead of --source, or in addition to it.

vpodzime commented on src/sbin/lorax:
```diff
@@ -302,6 +325,24 @@ def get_dnf_base_object(installroot, repositories, mirrorlists=None,
+            log.warning("%s is an unknown repo, not enabling it", r)
+        else:
+            log.info("Enabled repo %s", r)
+            repolist.enable()
```
I think the above two lines should be swapped. Otherwise, one could e.g. get a traceback from ``repolist.enable()`` with *Enabled repo something* already in the log. The same applies to the other for cycle below.

vpodzime commented:
Looks good to me otherwise.

bcl commented:
Thanks, that actually cannot happen since it does the lookup first, but I realized I was checking r not repolist like it should have been. fixed that and moved the log anyway.
